src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE_OF_ARRAY = 10;
+    private static final double GROWING_VALUE = 1.5;
+    private int sizeOfList;
+    private T[] arrayList;
+
+    public ArrayList() {
+        this.arrayList = (T[]) new Object[SIZE_OF_ARRAY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (arrayList.length == sizeOfList) {
+            listCreator();
+        }
+        arrayList[sizeOfList] = value;
+        sizeOfList++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > sizeOfList) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound Exception");
+        }
+        if (sizeOfList == arrayList.length) {
+            listCreator();
+        }
+        System.arraycopy(arrayList, index, arrayList, index + 1, sizeOfList - index);
+        arrayList[index] = value;
+        sizeOfList++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= sizeOfList || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound Exception");
+        }
+        return arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= sizeOfList || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound Exception");
+        }
```

Comment:
this checking is the same at a lot of methods, let's move it to the separate method: checkIndex(int index, int maxEvailableSize)